### ISSUE TITLE
Implement TextLineSpanners

### ISSUE THREAD

**Issue by ajyoon** [ajyoon/brown#8]
Several common notation primitives involve a bit of text combined with a (usually dashed) line, such as octave spanners and pedal markings. A compound `ObjectGroup` will be needed, consisting of a `TextObject` (or `MusicTextObject`?) and a `Path`. It will also be needed to implement different line stroke styles to achieve dashed lines, so some modification of `Pen` may be needed.

The init pattern for `TextLineSpanner` may look something like:

```python
TextLineSpanner(pos=(Mm(20), Mm(-3)), 
                length=Mm(50),
                text='some text',
                parent=some_staff)
```

Then the init pattern for a common marking like octave markings might be:

```python
OctaveSpanner(pos=(Mm(20), Mm(-3)), 
              length=Mm(50),
              octave=-1,  # Maybe use integer for different octaves? 
                          # -1 would be 8vb, 1 -> 8va, 2 -> 15ma, etc.
              parent=some_staff)
```

### TODO: 

- [x] Implement stroke patterns in `Pen`
- [x] Implement `RepeatingMusicTextLine`
- [x] Implement `OctaveLine`
- [x] Implement `PedalLine`

**Comment by ajyoon** [comment]
Pen stroke patterns implemented in 94b177af4ec5660575c5f4959196be2345970e62

**Comment by ajyoon** [comment]
`RepeatingMusicTextLine` base class implemented in 3fc090a7eb754b3ab0307d1a9c280c5654041f54. This implementation makes it easy to make subclasses or one-off `RepeatingMusicTextLine` objects:

```python
fake_trill = RepeatingMusicTextLine((Mm(30), staff.unit(-6), 0, staff),
                                    (Mm(500), staff.unit(-6), 0, staff),
                                    'wiggleTrill')
```

**Comment by ajyoon** [comment]
`OctaveLine` implemented in 7ef255367a58c5da1292202e8d34ed94528439c9

From a `TODO` in the new class:
The dashed line portion of this spanner overlaps with the '8va' text. This is a nontrivial fix that may require implementing text background masking or a way to easily inject line continuation offsets for paths.

**Comment by ajyoon** [comment]
Note also that music transposition under `OctaveLine` spanners is not implemented either.

**Comment by ajyoon** [comment]
`OctaveLine` transposition implemented in 74a9a4898272acb347014298f46371c3467fee63

**Comment by ajyoon** [comment]
2 most common forms of pedal lines implemented:
* ornate _Ped_ ... release star implemented as `PedAndStar` in 3de45ee34e8234bf5554beee6b440a79b05ada55
* Line with half-lift crooks implemented as `PedalLine` in 288c79e04997daa922c5c446c79bce5cb213547a
